### DISCUSSION TITLE
Is calling api through SWR is faster than calling through getserversideprops because of cache?

### DISCUSSION THREAD

**Discussion by roker15** [vercel/swr#2209] [Q&A]
I have to make decisions calling my api in getserversideprops or in Swr , however i feel that getserversideprops looks slower because swr immediately return data from cache. What approach o shud take

**Comment by dev-afzalansari** [comment]
@roker15,

> however i feel that getserversideprops looks slower because swr immediately return data from cache. What approach o shud take

it depends on the type of page you are building `getServerSideProps` enables ssr in your application for every page request while `swr` is for data fetching on the client side (also can be [paired with ssr](https://swr.vercel.app/docs/with-nextjs#pre-rendering-with-default-data)).

*  If data on your page changes frequently like a **User Dashboard** and ssr is not needed you should use `swr` only.
*  if the data changes frequently and ssr is needed like a **News Page** you should use `getServerSideProps`.

**Reply by roker15** [reply]
> @roker15,
> 
> > however i feel that getserversideprops looks slower because swr immediately return data from cache. What approach o shud take
> 
> it depends on the type of page you are building `getServerSideProps` enables ssr in your application for every page request while `swr` is for data fetching on the client side (also can be [paired with ssr](https://swr.vercel.app/docs/with-nextjs#pre-rendering-with-default-data)).
> 
> * If data on your page changes frequently like a **User Dashboard** and ssr is not needed you should use `swr` only.
> * if the data changes frequently and ssr is needed like a **News Page** you should use `getServerSideProps`.

Sorry u did not understand my question.. I did not ask when to use what. Is calling api in getserversideprops is slower than calling api in swr? Because swr immediately return data from cache but getserversideprops do not.

**Reply by dev-afzalansari** [reply]
@roker15

> What approach o shud take

Sorry, i tried to reply to this question specifically. Becuase elaborating uses & features of given strategies often leads to the answer.

> Is calling api in getserversideprops is slower than calling api in swr? Because swr immediately return data from cache but getserversideprops do not.

But for your this question, yes `swr` will return data fast but not for the first api call made by `swr` because there will be no `cache` stored at that time `swr` will store a `cache` after the first request which will take same amount of time as `getServerSideProps`(if same api) and then for further request it will return data from `cache`.
